python/cudf/cudf/pylibcudf_tests/common/utils.py

```diff
@@ -135,8 +145,80 @@ def is_fixed_width(plc_dtype: plc.DataType):
     )
 
 
+def nesting(typ) -> tuple[int, int]:
+    """Return list and struct nesting of a pyarrow type."""
```

Comment:
Maybe `nesting_level`? Just `nesting` could be returning a boolean, or it could be generating nesting, or any number of other things.